bumble/profiles/aics.py

```diff
@@ -0,0 +1,482 @@
+                    "GainMode should be either MANUAL or MANUAL_ONLY Cf Spec Audio Input Control Service 3.5.2.1"
+                )
+                return
+
+            if (
+                gain_settings_operand
+                < self.gain_settings_properties.gain_settings_minimum
+                or gain_settings_operand
+                > self.gain_settings_properties.gain_settings_maximum
+            ):
+                logger.error("gain_seetings value out of range")
+                raise ATT_Error(ErrorCode.VALUE_OUT_OF_RANGE)
+
+            if self.audio_input_state.gain_settings != gain_settings_operand:
+                self.audio_input_state.gain_settings = gain_settings_operand
+                await self.audio_input_state.notify_subscribers(connection)
+
+        async def _unmute():
+            '''Cf. 3.5.2.2 Unmute procedure'''
+
+            logger.error(f'unmute: {self.audio_input_state.mute}')
+            mute = self.audio_input_state.mute
+            if mute == Mute.DISABLED:
+                logger.error("unmute: Cannot change Mute value, Mute state is DISABLED")
+                raise ATT_Error(ErrorCode.MUTE_DISABLED)
+
+            if mute == Mute.NOT_MUTED:
+                return
+
+            self.audio_input_state.mute = Mute.NOT_MUTED
+            self.audio_input_state.increment_change_counter()
+            await self.audio_input_state.notify_subscribers(connection)
+
+        async def _mute() -> None:
+            '''Cf. 3.5.5.2 Mute procedure'''
+
+            change_counter = self.audio_input_state.change_counter
+            mute = self.audio_input_state.mute
+            if mute == Mute.DISABLED:
+                logger.error("mute: Cannot change Mute value, Mute state is DISABLED")
+                raise ATT_Error(ErrorCode.MUTE_DISABLED)
+
+            change_counter_operand = value[1]
+            if change_counter != change_counter_operand:
+                raise ATT_Error(ErrorCode.INVALID_CHANGE_COUNTER)
+
+            if mute == Mute.MUTED:
+                return
+
+            self.audio_input_state.mute = Mute.MUTED
+            self.audio_input_state.increment_change_counter()
+            await self.audio_input_state.notify_subscribers(connection)
+
+        async def _set_manual_gain_mode() -> None:
+            '''Cf. 3.5.2.4 Set Manual Gain Mode procedure'''
+
+            gain_mode = self.audio_input_state.gain_mode
+            if gain_mode in (GainMode.AUTOMATIC_ONLY, GainMode.MANUAL_ONLY):
+                logger.error(f"Cannot change gain_mode, bad state: {gain_mode}")
+                raise ATT_Error(ErrorCode.GAIN_MODE_CHANGE_NOT_ALLOWED)
+
+            if gain_mode == GainMode.MANUAL:
+                return
+
+            self.audio_input_state.gain_mode = GainMode.MANUAL
+            self.audio_input_state.increment_change_counter()
+            await self.audio_input_state.notify_subscribers(connection)
+
+        async def _set_automatic_gain_mode() -> None:
```

Comment:
I will prefer these functions to be member functions instead of inlines functions because:
1. Inline functions will be built on every visit, but only some of them are used on each call
2. Inline functions cannot be inherited